File: app/src/main/java/io/github/mohamedisoliman/androiddev/ui/base/AppBaseAdapter.java
package io.github.mohamedisoliman.androiddev.ui.base;

import android.support.v7.widget.RecyclerView;
import java.util.ArrayList;
import java.util.List;

/**
 * Created by Mohamed Ibrahim on 4/27/18.
 */
public abstract class AppBaseAdapter<VH extends RecyclerView.ViewHolder, T>
    extends RecyclerView.Adapter<VH> {

  private List<T> data = new ArrayList<>();

  public void setData(List<T> data) {
    this.data.clear();
    this.data.addAll(data);
    notifyDataSetChanged();
  }

  @Override
  public int getItemCount() {
    return data.size();
  }

  protected T getDataItem(int position) {
    return data.get(position);
  }

  public List<T> getData() {
    return data;
  }
}

File: app/src/main/java/io/github/mohamedisoliman/androiddev/ui/base/AppBaseAdapter.kt
package io.github.mohamedisoliman.androiddev.ui.base

import androidx.recyclerview.widget.RecyclerView
import java.util.ArrayList

/**
 * Created by Mohamed Ibrahim on 4/27/18.
 */
abstract class AppBaseAdapter<VH : androidx.recyclerview.widget.RecyclerView.ViewHolder, T> : androidx.recyclerview.widget.RecyclerView.Adapter<VH>() {

  val data = ArrayList<T>()

  fun setData(data: List<T>) {
    this.data.clear()
    this.data.addAll(data)
    notifyDataSetChanged()
  }

  override fun getItemCount(): Int {
    return data.size
  }

  protected fun getDataItem(position: Int): T {
    return data[position]
  }
}
